src/layer/BYOCLayer.ts

```diff
@@ -114,10 +114,21 @@ export class BYOCLayer extends AbstractSentinelHubV3Layer {
       }
 
       if (this.locationId === null) {
-        this.locationId =
-          (Object.entries(SHV3_LOCATIONS_ROOT_URL).find(
-            ([, url]) => url === this.shServiceRootUrl,
-          )?.[0] as LocationIdSHv3) ?? LocationIdSHv3.awsEuCentral1;
+        if (this.subType !== BYOCSubTypes.ZARR) {
+          const url = `${this.getSHServiceRootUrl()}api/v1/byoc/global/?ids=${this.collectionId}`;
```

Comment:
This endpoint probably doesn't return batch collections (undocumented, so we can't know), so I would change the if to `if (this.subType === BYOCSubTypes.BYOC) {` just for peace of mind.

Do you have more info about this endpoint?

Reply:
I double checked and it's ok to use this endpoint for all types. It even supports ZARRs, but I wouldn't change the logic here just yet as there might be other dependencies for those.